### PR TITLE
rgw: wrong x-rgw-quota-bucket-objects header when bucket quota is disabled

### PR DESCRIPTION
**Pull request by ArbitCode**
Fixes: https://tracker.ceph.com/issues/68211

Signed-off-by: Raja Sharma <raja@ibm.com>

Tracker: https://tracker.ceph.com/issues/68211

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x]  References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -2395,8 +2395,8 @@ static void dump_bucket_metadata(req_state *s, rgw::sal::Bucket* bucket,
   dump_header(s, "X-RGW-Object-Count", static_cast<long long>(stats.num_objects));
   dump_header(s, "X-RGW-Bytes-Used", static_cast<long long>(stats.size));
 
-  // only bucket's owner is allowed to get the quota settings of the account
-  if (s->auth.identity->is_owner_of(bucket->get_owner())) {
+  // only bucket's owner is allowed to get the quota settings of the account if quota is enabled
+  if (s->auth.identity->is_owner_of(bucket->get_owner()) && s->bucket->get_info().quota.enabled) {
```
Comment (cbodley) [inline src/rgw/rgw_rest_s3.cc:2399]: `s->bucket->get_info().quota.enabled` should only control whether we dump the Bucket-Size/Bucket-Objects headers

there's a separate `user_info.quota.user_quota.enabled` for the User-Size/User-Objects headers

the Max-Buckets header isn't controlled by either of those, so should always be dumped

**Comment by dang** [comment]
jenkins retest this please

File: src/rgw/rgw_rest_s3.cc (hunk 1 above)
Comment (ArbitCode) [reply src/rgw/rgw_rest_s3.cc]: @cbodley added enabled check for user and bucket info before dump headers : Bucket-Size/Bucket_Objects and User-Size/User-Objects.

Please review again.

**Review by dang — APPROVED** [review]

**Review by cbodley — APPROVED** [review]

File: src/mon/MonMap.cc [hunk 2]
```diff
@@ -196,7 +196,12 @@ void MonMap::encode(ceph::buffer::list& blist, uint64_t con_features) const
   if (!HAVE_FEATURE(con_features, MONENC) ||
       !HAVE_FEATURE(con_features, SERVER_NAUTILUS)) {
     for (auto& [name, info] : mon_info) {
-      legacy_mon_addr[name] = info.public_addrs.legacy_addr();
+      // see note in mon_info_t::encode()
+      auto addr = info.public_addrs.legacy_addr();
+      if (addr == entity_addr_t()) {
+        addr = info.public_addrs.as_legacy_addr();
+      }
+      legacy_mon_addr[name] = addr;
```
Comment (cbodley) [inline src/mon/MonMap.cc:204]: did you rebase and accidentally include this part?

**Review by cbodley — CHANGES_REQUESTED** [review]

File: src/mon/MonMap.cc (hunk 2 above)
Comment (ArbitCode) [reply src/mon/MonMap.cc]: I was trying rebase for ceph-ci so accidently above file included.
PR will fall on single logical commit.

**Comment by ArbitCode** [comment]
closing thi PR, since it messed up with history of commit.
New PR: https://github.com/ceph/ceph/pull/60756

**Pull request closed without merging**